community/values/src/main/java/org/neo4j/values/storable/NumberValues.java

```diff
@@ -36,15 +36,17 @@ private NumberValues()
 
     /*
      * Using the fact that the hashcode ∑x_i * 31^(i-1) can be expressed as
-     * a dot product, [v_1, v_2, v_2, ...] • [1, 31, 31^2,...]. By expressing
+     * a dot product, [1, v_1, v_2, v_2, ..., v_n] • [31^n, 31^{n-1}, ..., 31, 1]. By expressing
```

Comment:
`1, v_1, v_2, v_2` --> `1, v_1, v_2, v_3`?

Reply:
Yes the hashcode is for `a = [v_1, v_2,..., v_n]` is

```
 int result = 1;
        for (int element : a)
            result = 31 * result + element;
```
which is equivalent to

```
1 * 31^n + v_1 * 31^{n-1} + ... + v_{n-1} * 31 + v_n * 1 = [1, v_1,...,v_n] • [31^n, ..., 31^1, 31^0]
```